measured.go

```diff
@@ -273,12 +273,16 @@ func (mc *Conn) Close() (err error) {
 
 func (mc *Conn) submitTrafficIfNecessary() {
 	now := time.Now()
+	needToSubmit := false
 	mc.submitMutex.Lock()
 	if now.Sub(mc.lastSubmitted) > mc.interval {
-		mc.submitTraffic()
+		needToSubmit = true
 		mc.lastSubmitted = now
 	}
 	mc.submitMutex.Unlock()
+	if needToSubmit {
```

Comment:
Oh cool @oxtoacart. Yeah I was thinking of two separate locks (as in drop the read lock, grab the write lock if necessary), but maybe at the point all the locking itself becomes the biggest performance issue. This LGTM though. Nice change!